### PR TITLE
Adapt fileserver tests to handle IPv6 as well

### PR DESCRIPTION
**Pull request by benjamreis**
- escape IP address in regex as IPv6 have brackets in URL
- use local_command instead of direct call to Popen

### CODE REVIEW

File: tests/misc/test_file_server.py [hunk 1]
```diff
@@ -11,35 +12,27 @@
...
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE
-    )
-    stdout, _ = process.communicate()
-    lines = stdout.decode().splitlines()
-    assert lines[0].strip() == "HTTP/1.1 301 Moved Permanently"
-    assert _header_equal(lines[2], "location", "https://" + ip + path)
+    res = commands.local_cmd(["curl", "-i", "http://" + ip + path])
+    lines = res.stdout.splitlines()
+    assert lines[5].strip() == "HTTP/1.1 301 Moved Permanently"
```
Comment (stormi) [inline tests/misc/test_file_server.py:23]: Why line 5 now? What are the 5 new lines before it?

Comment (benjamreis) [reply tests/misc/test_file_server.py:23]:
```
% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
```

Don't know where it comes from.

Comment (stormi) [reply tests/misc/test_file_server.py:23]: An alias on your system?

**Comment by stormi** [comment]
I added @gthvn1 to the reviewers, as he wrote the test.

File: tests/misc/test_file_server.py (hunk 1 above)
Comment (benjamreis) [reply tests/misc/test_file_server.py]: solved by using `-s` in curl and avoiding the extra lines.

**Review by stormi — APPROVED** [review]

**Review by gduperrey — APPROVED** [review]
run pytest on the VM in ipv6: :white_check_mark:

**Pull request merged**